src/benchmarks/real-world/ImageSharp/Codecs/EncodeBmp.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Tests;
+
+namespace SixLabors.ImageSharp.Benchmarks.Codecs
+{
+    public class EncodeBmp
+    {
+        private Stream bmpStream;
+        private Image<Rgba32> bmpCore;
+
+        [GlobalSetup]
+        public void ReadImages()
+        {
+            if (this.bmpStream == null)
+            {
+                this.bmpStream = File.OpenRead(Path.Combine(TestEnvironment.InputImagesDirectoryFullPath, TestImages.Bmp.Car));
+                this.bmpCore = Image.Load<Rgba32>(this.bmpStream);
+                this.bmpStream.Position = 0;
+            }
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            this.bmpStream.Dispose();
+            this.bmpStream = null;
+            this.bmpCore.Dispose();
+        }
+
+        [Benchmark(Description = "ImageSharp Bmp")]
+        public void BmpImageSharp()
+        {
+            using var memoryStream = new MemoryStream();
```

Comment:
This stream should be a member, initialized in `[GlobalSetup]`, and `Seek`-ed to `0` before the `SaveAsBmp` call. We don't want the variance and the GC pressure stream allocation is causing here.

Reply:
But since `BmpImageSharp()` will be called repeatedly, won't it continue saving and growing the memory stream and increase memory pressure?

Reply:
If you seek to the beginning before saving, it only grows if you exceed the output size of the previous iteration, which is not the case, since you are saving a file of the same image deterministically. Or am I missing something?

Reply:
And that's the good thing. Are we are missing each others points?

This is what I mean:

```C#
private MemoryStream memoryStream;

[GlobalSetup]
public void Setup()
{
    this.memoryStream = new MemoryStream();
}

[Benchmark]
public void DoBenchmark()
{
    this.memoryStream.Seek(0, SeekOrigin.Begin); // avoid growing the buffer if we write the same amount of data over and over

   this.image.SaveAsBmp(this.memoryStream);
}
```

Reply:
Ah, I thought you wanted to do `this.memoryStream.Seek(0)` inside `GlobalSetup`. Will fix it. Thanks!